### DISCUSSION TITLE
Accessing pb_public via reverse-proxy throws CORS errors.

### DISCUSSION THREAD

**Discussion by ad-on-is** [pocketbase/pocketbase#4465] [Q&A] [answered]
I have a Nuxt app that has the following config

```ts
new pocketbase('http://127.0.0.1:8090')
```

When I build the app, I use the final product in `pb_public` so that pocketbase can serve it as a user-frontend.

The app is deployed via a Docker with the following command:

```Dockerfile
CMD ["/pocketbase", "serve", "--http=0.0.0.0:8090"]
```

Additionally I have a caddy server running as a reverse proxy.

```
example.com {
  reverse_proxy pocketbase:8090
}

```

Now, when I access `https://example.com/_` (the pocketbase admin dashboard) everything works fine. However, when I try to access anything else `https://example.com` I get CORS errors.

Am I doing something wrong?

![image](https://github.com/pocketbase/pocketbase/assets/9553253/c3db88c7-9383-488a-9d30-7e3996417245)

**Comment by ganigeorgiev** [comment]
PocketBase allows all origins (`*`) by default unless an explicit `--origins` flag is set, which doesn't seem to be in your case.

It is most likely something with the Caddy config. For an example how to configure CORS in Caddy you could check https://caddyserver.com/docs/caddyfile/directives/import#examples.

**Comment by ad-on-is** [comment]
I already have Caddy set up in a way to handle CORS, which works for all my services that I'm running behind it.

```
(defaults) {
  file_server
  encode gzip

  header {
    Access-Control-Allow-Origin "{http.request.header.origin}"
    Access-Control-Allow-Headers "{http.request.header.Access-Control-Request-Headers}"
    Access-Control-Expose-Headers "{http.request.header.Access-Control-Request-Headers}"
    Access-Control-Allow-Credentials true
    Access-Control-Allow-Methods "OPTIONS,HEAD,GET,POST,PUT,PATCH,DELETE"
    Vary Origin
  }
}

(options) {
  @options {
    method OPTIONS
  }

  handle @options {
    respond "OK" 204
  }
}

```

Additionally, I have this one set for each reverse-proxy

```
(reverse-proxy-settings) {
  header_down X-Real-IP {remote}
  header_down -Access-Control-Allow-Origin "{http.request.header.origin}"
  header_down -Access-Control-Allow-Headers "{http.request.header.Access-Control-Request-Headers}"
  header_down -Access-Control-Expose-Headers "{http.request.header.Access-Control-Request-Headers}"
  header_down -Access-Control-Allow-Credentials true
  header_down -Access-Control-Allow-Methods "OPTIONS,HEAD,GET,POST,PUT,PATCH,DELETE"
}
```

**Reply by ganigeorgiev** [reply]
Are you sure that the CORS issue is not because your Nuxt app can't access the PocketBase server via `http://127.0.0.1:8090`, aka. resulting in 404 for the `OPTIONS` call?

Note that your PocketBase Docker container has probably different localhost from your Nuxt dev app (_unless you run both applications in the same container_).

**Reply by ad-on-is** [reply]
I run both in the same container. As I already mentioned, the nuxt app is within **pb_public**, which is served by pocketbase itself.

**Reply by ganigeorgiev** [reply]
That's OK but if it is a static client-side app, aka. running in the browser, when you navigate to `https://example.com` the browser will send request to `http://127.0.0.1:8090`, not `https://example.com` and that will result to 404.

**Reply by ganigeorgiev** [reply, marked as answer]
If you don't want to specify a url when initializing the SDK you can use a relative base address, aka. `/` (or just `const pb = new PocketBase()`).

**Reply by ad-on-is** [reply]
Oh man... yes.. that was it... I totally forgot that the frontend app is accessing 127.0.0.1 (my PC)...dumb me. Thx again.
